lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -443,51 +445,82 @@ def _mtab_entries(self):
             mtab_entries.append(fields)
         return mtab_entries
 
-    @timeout.timeout()
+    def get_mount_info(self, mount_info, bind_mounts, uuids):
+
+            mount_info.update(get_mount_size(mount_info['mount']))
+
+            options = mount_info['options']
+            if mount_info['mount'] in bind_mounts:
+                # only add if not already there, we might have a plain /etc/mtab
+                if not self.MTAB_BIND_MOUNT_RE.match(options):
+                    options += ",bind"
+            mount_info['options'] = options
```

Comment:
This piece on mount should be moved to the outer level as it isn't in danger of timing out and will then always be provided to the caller (whereas if it's in here, it will be omitted if something else fails).